### ISSUE TITLE
SAML plugin internal error - ADFS

### ISSUE THREAD

**Issue by jesste** [bitium/jira-saml-plugin#36]
Hi. 
I am trying to configure the SAML plugin on my Jira (v6.0.8) to work together with my ADFS that is hosted in Azure. 

The redirect to my ADFS proxy's work but when I enter my AD credentials on my ADFS login page and are redirected back to Jira I get the "SAML plugin internal error" on a red bar in the top.

In the log file on Jira it says: 
**2016-04-18 14:52:41,998 http-bio-8080-exec-16 ERROR anonymous 892x7389x1 1e8fv8j 10.0.xx.xx,0:0:0:0:0:0:0:1 /plugins/servlet/saml/auth [bitium.saml.servlet.SsoLoginServlet] saml plugin error + Metadata for issuer https://sts.windows.net/f3b75253-4214-4a54-b052-56ac6906ae80/ wasn't found**

Not sure what to do from here.... Anyone had this problem?

**Comment by TheFlyingCorpse** [comment]
Yes. The issue is that you use one URL whereas the URL you see there is what you should actually configure for the trust URL.

**Comment by barbz127** [comment]
Can you please elaborate?
Im having similar issues as Jesste

On adfs
Identifier: https://jira.somecompany.com:443/jiraSAML
Endpoint https://jira.somecompany.com/plugins/servlet/saml/auth
Certificate loaded 
Note monitoring also fails hitting these two urls ADFS error 182180 - (404 not found)

On jira/addon
login url https://fs.somecompany.com/adfs/ls
UID NameID
Certificate (exists)
Entity ID https://fs.somecompany.com/adfs/services/trust

**Comment by wivaku** [comment]
We have a similar issue.  
When logging in with ADFS user we get on login page: "SAML plugin internal error".  
The log shows: `saml plugin error + Metadata for issuer http://adfs.myserver.com/adfs/services/trust wasn't found`. That is the URL that is used in the plugin field for Entity ID.
Not sure what is meant with the comment "the URL you see there is what you should actually configure for the trust URL".

**Comment by TheFlyingCorpse** [comment]
@wivaku Are you sure its http://adfs.myserver.com/adfs/services/trust and not https://adfs.myserver.com/adfs/services/trust or even https://adfs.myserver.com:443/adfs/services/trust

**Comment by wivaku** [comment]
Interesting, the setting in the add-on is `https://...` , the log msg is `http://`. In other systems/plug-ins we enter the `https://...` link. Will check.

**Comment by TheFlyingCorpse** [comment]
That is your ADFS configuration that is incorrect in its metadata.

**Comment by wivaku** [comment]
What do you mean exactly? That the ADFS "server" sends back "http" instead of "https" and that that is incorrect?

**Comment by TheFlyingCorpse** [comment]
Correct, check its properties in the AD FS mgmt console.
